tools/ds-identify

```diff
@@ -123,7 +123,7 @@ DS_MAYBE=2
 DI_DSNAME=""
 # this has to match the builtin list in cloud-init, it is what will
 # be searched if there is no setting found in config.
-DI_DSLIST_DEFAULT="MAAS ConfigDrive NoCloud AltCloud Azure Bigstep \
+DI_DSLIST_DEFAULT="MAAS ConfigDrive LXD NoCloud AltCloud Azure Bigstep \
```

Comment:
For the record, the order in ds-identify doesn't really matter as it is discovered based on the order configuration file on disk in the datasource_list config option in /etc/cloud/cloud.cfg or /etc/cloud/cloud.cfg.d/90_dpkg.cfg (included in cloud images), or cloudinit/settings.py or kernel_cmdline in the function [read_datasource_list](https://github.com/canonical/cloud-init/blob/main/tools/ds-identify#L567)

you can see the ds-identify sources this order in /run/cloud-init/ds-identify.log even if LXD is listed before NoCloud in ./tools/ds-identify as I had here... you'll still see datasource_list sourced.
```bash
/etc/cloud/cloud.cfg.d/90_dpkg.cfg set datasource_list: [ NoCloud, ConfigDrive, LXD, OpenNebula, DigitalOcean, Azure, AltCloud, OVF, MAAS, GCE, OpenStack, CloudSigma, SmartOS, Bigstep, Scaleway, AliYun, Ec2, CloudStack, Hetzner, IBMCloud, Oracle, Exoscale, RbxCloud, UpCloud, VMware, Vultr, None ]
DSLIST=NoCloud ConfigDrive LXD OpenNebula DigitalOcean Azure AltCloud OVF MAAS GCE OpenStack CloudSigma SmartOS Bigstep Scaleway AliYun Ec2 CloudStack Hetzner IBMCloud Oracle Exoscale RbxCloud UpCloud VMware Vultr None
check for 'NoCloud' returned found
check for 'LXD' returned found
Found 2 datasources found=all: NoCloud LXD
```
That ordering found by ds-identify sets the ordering of python datasources discovery:
 ```
root@dev-bb:~# grep LXD /var/log/cloud-init.log 
2021-10-30 02:34:33,682 - __init__.py[DEBUG]: Looking for data source in: ['NoCloud', 'LXD', 'None'], via packages ['', 'cloudinit.sources'] that matches dependencies ['FILESYSTEM']
```

All said though, it's best if we keep this ordering aligned. Because in absence of other defaults, eventually we'd fallback to ds-itentify ordering in the DI_DSLIST_DEAFULT.